File: C++/Data Structures/reversell.cpp
// Program to reverse a linked list
#include<iostream>
using namespace std;

class Node{
 public:
    int data;
    Node *next;
};

int main(){
    // prologic
    Node *head;
    Node *n1=new Node();
    Node *n2=new Node();
    Node *n3=new Node();
    Node *n4=new Node();
    Node *n5=new Node();

    n1->data=10;
    n2->data=20;
    n3->data=30;
    n4->data=40;
    n5->data=50;

    n1->next=n2;
    n2->next=n3;
    n3->next=n4;
    n4->next=n5;
    n5->next=NULL;

    head=n1;

    Node *prev=new Node();
    Node *curr=new Node();
    Node *nex=new Node();
    curr=head;
    while(curr!=NULL){
        nex=curr->next;
        curr->next=prev;
        prev=curr;
        curr=nex;
    }
    head=prev;
    while(head!=NULL){
        cout<<head->data<<" ";
        head=head->next;
    }






}

File: Python/Data Structures/reverseLL.py
class Node:
    def __init__(self,data=None):
        self.data=data
        self.next=None

n1=Node(11)
n2=Node(21)
n3=Node(31)
n4=Node(41)
n5=Node(51)
n6=Node(61)

n1.next=n2
n2.next=n3
n3.next=n4
n4.next=n5
n5.next=n6
n6.next=None

prev=None
curr=n1
next=n1.next

while curr is not None:
    curr.next=prev
    prev=curr
    curr=next
    if next is not None:
        next=next.next

head=prev

while head is not None:
    print(head.data,end=" ")
    head=head.next
print()